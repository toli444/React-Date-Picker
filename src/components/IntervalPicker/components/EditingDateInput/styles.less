.editing-date-input {
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;

  > span {
    margin-left: 10px;

    &:first-child {
      margin-left: 0;
    }
  }

  .selected {
    font-weight: bold;
  }
}
