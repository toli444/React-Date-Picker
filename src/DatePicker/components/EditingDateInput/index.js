import React from 'react';
import moment from 'moment'
import styles from './styles.less';

export default function EditingDateInput({currentDate, editingUnit, format}) {
  const month = moment(currentDate, format).format('MMM');
  const year = moment(currentDate, format).format('YYYY');

  return <div className={styles['editing-date-input']}>
    <span className={editingUnit === 'month' ? styles.selected : ''}>{month}</span>
    <span className={editingUnit === 'year' ? styles.selected : ''}>{year}</span>
  </div>;
};
