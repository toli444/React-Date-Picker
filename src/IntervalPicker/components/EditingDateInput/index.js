import React from 'react';
import moment from 'moment'
import PropTypes from 'prop-types';

import styles from './styles.less';

const EditingDateInput = ({currentDate, editingUnit, format}) => {
  const month = moment(currentDate, format).format('MMM');
  const year = moment(currentDate, format).format('YYYY');

  return <div className={styles['editing-date-input']}>
    <span className={`${editingUnit === 'month' ? styles.selected : ''} month`}>{month}</span>
    <span className={`${editingUnit === 'year' ? styles.selected : ''} year`}>{year}</span>
  </div>;
};

EditingDateInput.propTypes = {
  currentDate: PropTypes.object,
  editingUnit: PropTypes.string,
  format: PropTypes.string,
};

export default EditingDateInput;
