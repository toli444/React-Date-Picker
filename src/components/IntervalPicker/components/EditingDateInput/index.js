import React from 'react';
import moment from 'moment'
import PropTypes from 'prop-types';

import styles from './styles.less';

const EditingDateInput = ({currentDate, editingUnit, format}) => {
  const month = currentDate && moment(currentDate, format).format('MMM');
  const year = currentDate && moment(currentDate, format).format('YYYY');

  return <div className={styles['editing-date-input']}>
    {month && <span className={`${editingUnit === 'month' ? styles.selected : ''} month`}>{month}</span>}
    {year && <span className={`${editingUnit === 'year' ? styles.selected : ''} year`}>{year}</span>}
  </div>;
};

EditingDateInput.propTypes = {
  currentDate: PropTypes.object,
  editingUnit: PropTypes.string,
  format: PropTypes.string,
};

export default EditingDateInput;
