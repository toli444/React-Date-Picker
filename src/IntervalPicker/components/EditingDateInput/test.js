import React from 'react';
import EditingDateInput from '../EditingDateInput';
import { shallow } from 'enzyme';

jest.mock('moment', () => () => ({format: () => 'today'}));

describe('EditingDateInput', () => {
  afterEach(() => {
    jest.resetAllMocks()
  });

  it('renders correctly', () => {
    const wrapper  = shallow(<EditingDateInput />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with currentDate', () => {
    const wrapper  = shallow(<EditingDateInput currentDate={'today'}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('adds selected class to editing month unit node', () => {
    const wrapper = shallow(<EditingDateInput editingUnit="month" currentDate={'today'}/>);

    expect(wrapper.find('.month').hasClass('selected'));
    expect(wrapper.find('.year').hasClass('selected')).toBeFalsy();
  });

  it('adds selected class to editing year unit node', () => {
    const wrapper = shallow(<EditingDateInput editingUnit="year" currentDate={'today'}/>);

    expect(wrapper.find('.month').hasClass('selected')).toBeFalsy();
    expect(wrapper.find('.year').hasClass('selected'));
  });

  it('doesn\'t add selected class if unit is not specified', () => {
    const wrapper = shallow(<EditingDateInput currentDate={'today'}/>);

    expect(wrapper.find('.month').hasClass('selected')).toBeFalsy();
    expect(wrapper.find('.year').hasClass('selected')).toBeFalsy();
  });
});
