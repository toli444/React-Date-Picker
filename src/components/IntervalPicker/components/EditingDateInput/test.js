import React from 'react';
import EditingDateInput from './index';
import { shallow } from 'enzyme';
import moment from 'moment';

const currentDate = moment('28/07/1997', 'DD/MM/YYYY');

describe('EditingDateInput', () => {
  it('renders correctly', () => {
    const wrapper  = shallow(<EditingDateInput />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with currentDate', () => {
    const wrapper  = shallow(<EditingDateInput currentDate={currentDate}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('adds selected class to editing month unit node', () => {
    const wrapper = shallow(<EditingDateInput editingUnit="month" currentDate={currentDate}/>);

    expect(wrapper.find('.month').hasClass('selected')).toBeTruthy();
    expect(wrapper.find('.year').hasClass('selected')).toBeFalsy();
  });

  it('adds selected class to editing year unit node', () => {
    const wrapper = shallow(<EditingDateInput editingUnit="year" currentDate={currentDate}/>);

    expect(wrapper.find('.month').hasClass('selected')).toBeFalsy();
    expect(wrapper.find('.year').hasClass('selected')).toBeTruthy();
  });

  it('doesn\'t add selected class if unit is not specified', () => {
    const wrapper = shallow(<EditingDateInput currentDate={currentDate}/>);

    expect(wrapper.find('.month').hasClass('selected')).toBeFalsy();
    expect(wrapper.find('.year').hasClass('selected')).toBeFalsy();
  });
});
