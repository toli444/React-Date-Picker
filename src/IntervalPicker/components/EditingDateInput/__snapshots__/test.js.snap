// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditingDateInput renders correctly 1`] = `
<div
  className="editing-date-input"
/>
`;

exports[`EditingDateInput renders correctly with currentDate 1`] = `
<div
  className="editing-date-input"
>
  <span
    className=" month"
  >
    today
  </span>
  <span
    className=" year"
  >
    today
  </span>
</div>
`;
